google/cloud/spanner_v1/_opentelemetry_tracing.py

```diff
@@ -15,41 +15,89 @@
 """Manages OpenTelemetry trace creation and handling"""
 
 from contextlib import contextmanager
+import os
 
 from google.api_core.exceptions import GoogleAPICallError
 from google.cloud.spanner_v1 import SpannerClient
+from google.cloud.spanner_v1 import gapic_version as TRACER_VERSION
 
 try:
     from opentelemetry import trace
     from opentelemetry.trace.status import Status, StatusCode
+    from opentelemetry.semconv.trace import SpanAttributes
 
     HAS_OPENTELEMETRY_INSTALLED = True
+    DB_SYSTEM = SpanAttributes.DB_SYSTEM
+    DB_NAME = SpanAttributes.DB_NAME
+    DB_CONNECTION_STRING = SpanAttributes.DB_CONNECTION_STRING
+    NET_HOST_NAME = SpanAttributes.NET_HOST_NAME
+    DB_STATEMENT = SpanAttributes.DB_STATEMENT
 except ImportError:
     HAS_OPENTELEMETRY_INSTALLED = False
 
 
+EXTENDED_TRACING_ENABLED = os.environ.get('SPANNER_ENABLE_EXTENDED_TRACING', '') == 'true'
+
+SPAN_NAMESPACE = 'cloud.google.com/python/spanner'
+TRACER_NAME = SPAN_NAMESPACE
+
+def get_tracer(tracer_provider=None):
+    """
+    get_tracer is a utility to unify and simplify retrieval of the tracer, without
+    leaking implementation details given that retrieving a tracer requires providing
+    the full qualified library name and version.
+    When the tracer_provider is set, it'll retrieve the tracer from it, otherwise
+    it'll fall back to the global tracer provider and use this library's specific semantics.
+    """
+    if tracer_provider:
+        return tracerProvider.get_tracer(TRACER_NAME, TRACER_VERSION)
```

Comment:
I tried to run one sample and see that `otel.scope.name` is not getting populated.
![image](https://github.com/user-attachments/assets/a9a7cb93-6046-4b3f-9751-9d72df3163a5)

It should look something like below (this was in java)
![image](https://github.com/user-attachments/assets/36cfc5bc-a937-40c4-9dc9-46ef1ba9e2a3)